File: polygot_playground/python/dsa/array/permuteArray.mjs
// //node --experimental-modules permuteArray.mjs
// // Input: [1,2,3]
// // Output: [[1,2,3],[1,3,2],[2,1,3],[2,3,1],[3,1,2],[3,2,1]]

const permute = function(nums) {
    let results = [];
    let go = (current) => {
      if (current.length === nums.length){
        results.push(current);
        return;
      }
      nums.forEach(n => {
        if (!current.includes(n)){
            console.log(current);
            console.log(n);
          go([...current, n]);
        }
      });
    }
    go([]);
    return results;
};

permute([1,2,3]);
//console.log(permute([1,2,3]));

File: polygot_playground/python/dsa/array/permuteArray.py
# // Input: [1,2,3]
# // Output: [[1,2,3],[1,3,2],[2,1,3],[2,3,1],[3,1,2],[3,2,1]]

def permute(array):
  results = []
  def go(current):
    if(len(current) == len(array)):
      results.append(current)
      return
    for n in array:
      if not (n in current):
        temp = [t for t in current]
        temp.append(n)
        go(temp)
  
  go([])
  return results

print(permute([1,2,3]));